test/__init__.py

```diff
@@ -139,6 +140,19 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def requires_bundled_OpenSSL_on_mac(test):
```

Comment:
Could we change the name of this decorator and make it an xfail instead of a skip? Also let's be clearer on where this fails, so far we've seen that it fails on macOS with OpenSSL < 1.0.0? Let's test for that rather than a specific Python version.

Reply:
Thanks, fixed. It's not fun to parse the version robustly in a string like "OpenSSL 0.9.8zh 14 Jan 2016" and I guess it's better to be too specific than too broad so I now simply match that 0.9.8zh string.